cranelift/codegen/src/machinst/compile.rs

```diff
@@ -26,26 +26,24 @@ where
     debug!("vcode from lowering: \n{}", vcode.show_rru(Some(universe)));
 
     // Perform register allocation.
-    let algorithm = match vcode.flags().regalloc() {
-        settings::Regalloc::Backtracking => RegAllocAlgorithm::Backtracking,
-        settings::Regalloc::BacktrackingChecked => RegAllocAlgorithm::BacktrackingChecked,
-        settings::Regalloc::ExperimentalLinearScan => RegAllocAlgorithm::LinearScan,
+    let algorithm_with_defaults = match vcode.flags().regalloc() {
+        settings::Regalloc::Backtracking => AlgorithmWithDefaults::Backtracking,
+        settings::Regalloc::BacktrackingChecked => AlgorithmWithDefaults::Backtracking,
```

Comment:
This will break checking with when using the backtracking-checked variant; we should probably use the `allocate_registers_with_opts` here instead, passing `run_checker: true` with this variant.